### DISCUSSION TITLE
Disabling of GFF parsing step of "_sp_" scripts

### DISCUSSION THREAD

**Discussion by Carolinacanettieri** [NBISweden/AGAT#431] [Q&A]
Hi,

I am using AGAT to extract different sequences from genomes by taking as input the GFF2 file that Exonerate generates. In this process, before using AGAT's `agat_sp_extract_sequences.pl`, I attribute an uniq ID for each gene Exonerate identifies. I do this by manually creating an "ID=" tag, which is posteriorly transformed into the "gene_id" tag by AGAT's `agat_convert_sp_gff2gtf.pl`. Adding this uniq ID works perfectly. However, the parsing step that AGAT automatically performs in its _sp_ scripts ends up "creating new genes". For what I need, the Exonerate's GFF2 file has sufficient information to extract the sequences it identifies. Is there a way I can disable the parsing step of _sp_ scripts? Or does anyone have suggestions on how to do deal with this problem?

**Comment by Juke34** [comment]
You can deactivate some steps in via the config file (at the end of the file) `agat config --expose`, but it is not recommended. 
AGAT is made to standardise the annotation, so it is why it add the gene feature if it is missing in your file. You should no pay attention to it if you do not need it. This behavior cannot be deactivated because each feature should be attached to parent feature until the higher level which is level1. The only way to achieve what you want  is to modify the feature_levels.yaml to have mRNA as feature level1 and put exon, CDS, etc as level 2, and not have any feature level3. You can do that by just renaming files level2 -> level1 and level3 to level2 and removing the level1 one.
